test/gtest/uct/test_md.cc

```diff
@@ -303,11 +305,20 @@ UCS_TEST_P(test_md, mem_type_detect_mds) {
                               UCT_MD_MEM_ATTR_FIELD_SYS_DEV      |
                               UCT_MD_MEM_ATTR_FIELD_BASE_ADDRESS |
                               UCT_MD_MEM_ATTR_FIELD_ALLOC_LENGTH;
-        status = uct_md_mem_query(md(), address, buffer_size, &mem_attr);
-        ASSERT_UCS_OK(status);
-        EXPECT_EQ(alloc_mem_type, mem_attr.mem_type);
-        EXPECT_TRUE((mem_attr.alloc_length >= buffer_size) || (mem_attr.alloc_length == -1));
-        EXPECT_TRUE((mem_attr.base_address == NULL) || (mem_attr.base_address = address));
+
+        for (unsigned i = 0; i < 300; i++) {
+            offset        = ucs::rand() % buffer_size;
+            length_offset = ucs::rand() % buffer_size;
+
+            status = uct_md_mem_query(md(), (void*)((char*)address + offset),
+                                      length_offset, &mem_attr);
+            ASSERT_UCS_OK(status);
+            EXPECT_EQ(alloc_mem_type, mem_attr.mem_type);
+            EXPECT_TRUE((mem_attr.alloc_length >= buffer_size) ||
```

Comment:
why not == ?

Reply:
I was concerned about the case where the underlying allocator may decide to allocate larger range (say because it only allocates at page granularity). I can change this to `==` as it would work for CUDA.

Reply:
we can adjust our expectation according to the memory type

Reply:
I've changed to use `==` in the PR.